DocsScraper/src/DocsScraper.jl

```diff
@@ -0,0 +1,152 @@
+"""
+    get_base_url(url::String)
+
+Extracts the base url.
+
+# Arguments
+- `url`: The url string of which, the base url needs to be extracted
+"""
+function get_base_url(url::String)
+    parsed_url = URIs.URI(url)
+    base_url = string(parsed_url.scheme, "://", parsed_url.host, parsed_url.port != nothing ? ":" * string(parsed_url.port) : "", parsed_url.path)
+    return base_url
+end
+
+
+"""
+    process_nodes(node, heading_hierarchy::Dict, parsed_text::Vector)
+
+Process the nodes recursively. The function recursively calls itself for every HTML hierarchy, thereby going deeper to retrieve the text.
+
+# Arguments
+- `node`: The root HTML node 
+- `heading_hierarchy`: Dictionary used to store metadata
+- `parsed_text`: Vector of Dicts to store parsed text and metadata
+
+# TODO:
+- Inline code blocks are the ones present inside <p> or <li>. put `` around the inline code blocks. Make changes inside `elseif tag_name == "p" || tag_name == "li"`
+"""
+function process_nodes(node, heading_hierarchy::Dict, parsed_text::Vector)
```

Comment:
> What's the thinking behind providing `heading_hierarchy` and `parsed_text`? Is it to incrementally add to the same objects when scanning multiple URLs?
> 
> In general, unless there are huge performance benefits, it will be easier to work with if the `output` containers are initialized inside of the function AND have proper types, eg, you know that the certain fields will always be string, etc, so we do not need to use the generic containers with "Any" type, eg, `Dict()` is effectively `Dict{Any,Any}()` -> We probably know more about its types?
> 
> If you want to have `process_nodes` incrementally fill these output containers, the convention is to put "!" in the function name to indicate that it's a mutating function -> `process_nodes!` - see https://docs.julialang.org/en/v1/manual/style-guide/#bang-convention

So in `Dict`, the key is always `String` so I'll specify `String` but value could either be a `String` or a `Dict`, so should I use `Any`?